.travis/yapf.sh

```diff
@@ -52,13 +51,13 @@ format_changed() {
 
     if ! git diff --diff-filter=ACM --quiet --exit-code "$MERGEBASE" -- '*.py' &>/dev/null; then
         declare -a unformatted_files && mapfile -t unformatted_files < <(git diff --name-only --diff-filter=ACM "$MERGEBASE" -- '*.py')
-        yapf "${YAPF_EXCLUDES[@]}" "${YAPF_FLAGS[@]}" -- "${unformatted_files[@]}"
+        yapf --in-place "${YAPF_EXCLUDES[@]}" "${YAPF_FLAGS[@]}" -- "${unformatted_files[@]}"
     fi
 }
 
-# Format all files
+# Format all files, and print the diff to stdout for travis.
 format_all() {
-    yapf "${YAPF_FLAGS[@]}" "${YAPF_EXCLUDES[@]}" test python
+    yapf --diff "${YAPF_FLAGS[@]}" "${YAPF_EXCLUDES[@]}" test python
```

Comment:
This works pretty much as you'd expect it to, though I dunno if printing a diff is all that helpful.

It makes this function useless as a pre-push hook since you end up having to apply the formatting changes manually.

The files it's run over are the ones that are supposed to be autoformatted already, and currently are in the codebase. Files that have been formatted with yapf that have been checked in to master should from then on always be formatted before being merged in. If we insist on that, over time, we could phase out all the complicated mergebase checking and just format changed file.

Reply:
This is run by travis. What happens is that the user gets a report that some files aren't formatted, with no feedback as to what the problem is!!

Reply:
We will never be able to drop the mergebase, as that is necessary for speed.

Reply:
It *should* print out the files in the last block, with the `git --no-pager diff --name-only`.

We actually could drop the mergebase at some point and just run it over files modified in the last commit once the whole project has moved towards autoformatting. Since it would only touch the modified files, it'd be plenty fast.

Reply:
That's not the issue, without the mergebase you'll have the wrong set of changed files, period.

Reply:
@alok To give you context about our workflow: often times we're modifying our source code on a fresh clone (different instance). Setting up a pre-push every time doesn't make sense for us. 

This thing is primarily run on Travis (hence it's in the `.travis` folder), and we modify our files by looking at the tests/console output. That's why printing a diff is extremely helpful. 

Also, this means `git --no-pager diff --name-only` doesn't cut it. In fact, if we agree that this is for travis, we can consider dropping that line (L81 @ericl). 

Alternatively, if you want this as a prehook, you may consider adding a flag such that it satisfies your needs.